realm/src/main/java/io/realm/Realm.java

```diff
@@ -1872,4 +1890,24 @@ public static Object getDefaultModule() {
         void execute(Realm realm);
     }
 
+    /**
+     * Encapsulates an async {@link RealmQuery}.
+     * <p>
+     * This will run the {@link RealmQuery} on a worker thread, then invoke this callback on the caller thread
+     */
+    public interface AsyncCallback<T extends RealmResults<? extends RealmObject>> {
+        void onSuccess (T results);
+        void onError (Throwable t);
```

Comment:
While this is probably the right thing to do, it is also a bit dangerous as this will also return OutOfMemoryError and the like, and if people catch those and ignore them we are in a very bad place. I think @kneth is looking into how to handle these errors though.